Removed Excel export, added Devices Export button

lorthirk opened the pull request:
This PR removes Excel export in Data and Device Events views, and also removes the dependency from Apache POI that was needed for Excel exporting.

It also restores the Export button in Devices view.

**Related Issue**
This PR fixes #1981 

**Description of the solution adopted**
I just removed the Excel-related classes, and replaced dropdown buttons with plain buttons.

Also, the servlet for the Device Export button was already present; I only created the new button and pointed the click event to the servlet

**Screenshots**
N/A

**Any side note on the changes made**
N/A

lorthirk commented:
Hey @LeoNerdoG,

It would be great if you could test the Devices export checking if all the filters are correctly applied when creating the file 😄

LeoNerdoG commented:
Hey @lorthirk I have tested this. Bellow are my findings for Device view (split in main list view and Event view) and for Data view. 
**_CSV from Device view:_**
1. When downloading Device list, there are no Connection status, no Last Event On and no Last Event Type data in the CSV file (Screenshot No. 1). I don't know if this is OK or not as this may be sesnitive information? 
2. The title of the document consists ID of the child account, which is inpractical for end user. I recommend changing this to child account name (Screenshot No. 2)
3. I would change the first two field names from "id" and "Account" to "Account Id" and "Account Name" for clarification (Screenshot No. 3) 

_**CSV from Device Events:**_
1.There are columns "Created By" and "Device ID" - could you change this to user friendly names? (Screenshot No. 4).
2. Event message is not completely visible (Screenshot No. 5 and No. 6)
3. Title of the document is not descriptive enaugh (in my case device in child0 account was called "aaa" and the title was child0-device-events-4. Look at Screenshot No. 7)


_**CSV from Data View:**_
1. If Date Range is set so nod ata is shown, Export Data button is disabled, although in the same scenario in Devices (if no devices are present) the Export button is enabled. Is this OK? Because although no data is in the list, this empty list can be a evidence, proof of empty Dataview? (Screenshot No. 8)


**Screenshot No. 1:**
<img width="949" alt="screen shot 2018-09-11 at 09 48 03" src="https://user-images.githubusercontent.com/15121999/45345962-26823200-b5a8-11e8-888a-033f472b85cf.png">

**Screenshot No. 2:**
![screen shot 2018-09-11 at 09 58 13](https://user-images.githubusercontent.com/15121999/45346407-6564b780-b5a9-11e8-82da-9ac781b41438.png)

**Screenshot No. 3:**
![screen shot 2018-09-11 at 10 04 03](https://user-images.githubusercontent.com/15121999/45346647-0a7f9000-b5aa-11e8-8f76-17d8034fa878.png)

**Screenshot No. 4:**
<img width="850" alt="screen shot 2018-09-11 at 10 17 38" src="https://user-images.githubusercontent.com/15121999/45347268-ee7cee00-b5ab-11e8-8ce8-6b2652a5b050.png">

**Screenshot No. 5 (what is shown in the CVS document):**
<img width="720" alt="screen shot 2018-09-11 at 10 19 03" src="https://user-images.githubusercontent.com/15121999/45347345-1f5d2300-b5ac-11e8-946b-359e37076e7c.png">

**Screenshot No. 6 (what should actually be shown in CVS document):**
![screen shot 2018-09-11 at 10 19 34](https://user-images.githubusercontent.com/15121999/45347405-4ca9d100-b5ac-11e8-9ecc-730d3ce1194f.png)

**Screenshot No. 7:**
![screen shot 2018-09-11 at 10 44 29](https://user-images.githubusercontent.com/15121999/45348711-ad86d880-b5af-11e8-8474-7566d8a5cc7a.png)

**Screenshot No. 8:**
![screen shot 2018-09-11 at 10 58 23](https://user-images.githubusercontent.com/15121999/45349455-9d6ff880-b5b1-11e8-9d38-8ff06d9e19e0.png)

lorthirk commented:
I addressed all your concerns apart except the following ones:

> **_CSV from Device view:_**
> 1.There are columns "Created By" and "Device ID" - could you change this to user friendly names? (Screenshot No. 4).

The Device ID is... well, the ID  :) there's no "human" version of it; we could put in the Client ID instead but it would be redundant since it would never change; also, the Client ID is in the file name now

> 2. Event message is not completely visible (Screenshot No. 5 and No. 6)

This looks like an Excel issue. Try to import the CSV in Excel specifying all the correct format options and it should appear. Also, in Apple Numbers it gets correctly imported automatically.

LeoNerdoG commented:
Hey @lorthirk,
I have reviewed the changes and everything looks fine now except the last point: 

> **_CSV from Data View:_**
> 
> If Date Range is set so no data is shown, Export Data button is disabled, although in the same scenario in Devices (if no devices are present) the Export button is enabled. Is this OK? Because although no data is in the list, this empty list can be a evidence, proof of empty Dataview? (Screenshot No. 8)

Also I have another suggestion - the Timestamp column in Data CSV has "(UTC)" added so user knows the exact time when data was published. 
Can you add this also to:
 - Device's Created On column
 - Device's Last Event On column
 - Devices's Event Created On column
 - Devices's Event Received On column
 - Devices's Event Sent On column

I think this would be a good feature for end-user. 
Other than that I think it is OK.

lorthirk commented:
The Devices Export button should be disalbed when no devices are present. Can you confirm?

LeoNerdoG commented:
I am confirming that button is disabled when no devices are present in the device list.
